source/views/building-hours/report/editor.tsx

```diff
@@ -31,146 +32,127 @@ type Props = TopLevelViewPropsType & {
+		setSet(newSet)
+		props.route.params.onEditSet(newSet)
 	}
 
-	render() {
-		let set = this.state.set || blankSchedule()
-
-		let {open, close} = parseHours(set, moment())
-
-		return (
-			<ScrollView>
-				<TableView>
-					<Section header="DAYS">
-						<WeekToggles days={set.days} onChangeDays={this.onChangeDays} />
-					</Section>
-
-					<Section header="TIMES">
-						<DatePickerCell
-							date={open}
-							onChange={this.onChangeOpen}
-							title="Open"
-						/>
-						<DatePickerCell
-							date={close}
-							onChange={this.onChangeClose}
-							title="Close"
-						/>
-					</Section>
-
-					<Section>
-						<DeleteButtonCell onPress={this.delete} title="Delete Hours" />
-					</Section>
-				</TableView>
-			</ScrollView>
-		)
-	}
+	let {open, close} = parseHours(set, moment())
+
+	return (
+		<ScrollView>
+			<TableView>
+				<Section header="DAYS">
+					<WeekToggles days={set.days} onChangeDays={onChangeDays} />
+				</Section>
+
+				<Section header="TIMES">
+					<DatePickerCell date={open} onChange={onChangeOpen} title="Open" />
+					<DatePickerCell date={close} onChange={onChangeClose} title="Close" />
+				</Section>
+
+				<Section>
+					<DeleteButtonCell onPress={deleteSet} title="Delete Hours" />
+				</Section>
+			</TableView>
+		</ScrollView>
+	)
 }
 
 type WeekTogglesProps = {
 	days: DayOfWeekEnumType[]
-	onChangeDays: (days: DayOfWeekEnumType[]) => any
+	onChangeDays: (days: DayOfWeekEnumType[]) => unknown
 }
 
-class WeekTogglesIOS extends React.PureComponent<WeekTogglesProps> {
-	toggleDay = (day: DayOfWeekEnumType) => {
-		this.props.onChangeDays(xor(this.props.days, [day]))
-	}
-
-	render() {
-		let allDays = ['Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa', 'Su']
-
-		return (
-			<Row alignItems="stretch" justifyContent="center">
-				{allDays.map((day) => (
-					<ToggleButton
-						key={day}
-						active={this.props.days.includes(day)}
-						onPress={this.toggleDay}
-						text={day}
-					/>
-				))}
-			</Row>
-		)
-	}
+function WeekTogglesIOS(props: WeekTogglesProps) {
+	let {onChangeDays, days} = props
+
+	let toggleDay = useCallback(
+		(day) => {
+			onChangeDays(xor(days, [day]))
+		},
+		[onChangeDays, days],
+	)
+
+	let allDays: DayOfWeekEnumType[] = ['Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa', 'Su']
+
+	return (
+		<Row style={styles.weekToggles}>
+			{allDays.map((day) => (
+				<ToggleButton
+					key={day}
+					active={props.days.includes(day)}
+					onPress={toggleDay}
+					text={day}
+				/>
+			))}
+		</Row>
+	)
 }
 
-class WeekTogglesAndroid extends React.PureComponent<WeekTogglesProps> {
-	toggleDay = (day) => {
-		this.props.onChangeDays(xor(this.props.days, [day]))
-	}
+function WeekTogglesAndroid(props: WeekTogglesProps) {
+	let {onChangeDays, days} = props
 
-	render() {
-		let allDays = ['Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa', 'Su']
+	let toggleDay = useCallback(
+		(day) => {
+			onChangeDays(xor(days, [day]))
+		},
+		[onChangeDays, days],
+	)
```

Comment:
Could we move allDays and toggleDay out of the ios and android functional calls? I think they're identical for each platform.

Reply:
They are. Let's log that for future cleanup? Mainly because I'm away from the laptop for the evening, I think.

Reply:
Sounds good to me.